mobile/examples/phi-3-vision/android/android/app/src/main/java/ai/onnxruntime/genai/vision/demo/MainActivity.java

```diff
@@ -0,0 +1,351 @@
+            public void onClick(View v) {
+                if (model == null) {
+                    // if the edit text is empty display a toast message.
+                    Toast.makeText(MainActivity.this, "Model not loaded yet, please wait...", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                // Checking if the message entered
+                // by user is empty or not.
+                if (userMsgEdt.getText().toString().isEmpty()) {
+                    // if the edit text is empty display a toast message.
+                    Toast.makeText(MainActivity.this, "Please enter your message..", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                String promptQuestion = "<|user|>\n";
+                if (inputImage != null) {
+                    promptQuestion += "<|image_1|>\n";
+                }
+                promptQuestion += userMsgEdt.getText().toString() + "<system>You are a helpful AI assistant. Answer in two paragraphs or less<|end|>\n<|assistant|>\n";
+                final String promptQuestion_formatted = promptQuestion;
+
+                Log.i("GenAI: prompt question", promptQuestion_formatted);
+                setVisibility();
+
+                // Disable send button while responding to prompt.
+                sendMsgIB.setEnabled(false);
+
+                promptTV.setText(userMsgEdt.getText().toString());
+                // Clear Edit Text or prompt question.
+                userMsgEdt.setText("");
+                if (inputImage != null) {
+                    generatedTV.setText("[analyzing image...]\n");
+                }
+                else {
+                    generatedTV.setText("");
+                }
+
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        TokenizerStream stream = null;
+                        GeneratorParams generatorParams = null;
+                        Generator generator = null;
+                        Sequences encodedPrompt = null;
+                        Images images = null;
+                        NamedTensors inputTensors = null;
+                        try {
+                            stream = multiModalProcessor.createStream();
+
+                            generatorParams = model.createGeneratorParams();
+                            //examples for optional parameters to format AI response
+                            //generatorParams.setSearchOption("length_penalty", 1000);
+                            //generatorParams.setSearchOption("max_length", 500);
+
+                            if (inputImage != null) {
+                                images = inputImage.getImages();
+                            }
+
+
+                            inputTensors = multiModalProcessor.processImages(promptQuestion_formatted, images);
+                            generatorParams.setInput(inputTensors);
```

Comment:
had to update this to get it to build:
```suggestion
                            generatorParams.setInputs(inputTensors);
```